src/Core/Content/Newsletter/SalesChannel/SalesChannelNewsletterController.php

```diff
@@ -49,7 +49,7 @@ public function __construct(
      */
     public function subscribe(Request $request, RequestDataBag $data, SalesChannelContext $context): JsonResponse
     {
-        $data->set('storefrontUrl', $request->attributes->get('sw-sales-channel-absolute-base-url'));
+        $data->set('storefrontUrl', $request->attributes->get('sw-sales-channel-absolute-base-url') . $request->attributes->get('sw-sales-channel-base-url'));
```

Comment:
could you try `Shopware\Storefront\Framework\Routing\RequestTransformer::STOREFRONT_URL` ?
this should contain the whole URL

https://github.com/shopware/platform/blob/master/src/Storefront/Framework/Routing/RequestTransformer.php#L56